rfcs/0000-minimal-daemon.md

```diff
@@ -0,0 +1,67 @@
+---
+feature: minimal-daemon
+start-date: 2020-5-28
+author: John Ericson
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+`nix-daemon` should be a separate executable.
+
+# Motivation
+[motivation]: #motivation
+
+With flakes and other development, we are moving towards a more "batteries included" Nix command line.
+We don't want any of those features in the daemon, however, because the daemon is a special trusted process that we should strive to keep as simple as possible.
+\[This is comparable to a compilation pipeline, with a concise intermediate representation nicer user-facing features "desugar" into.\]
```

Comment:
Nit that I think slightly improves readability:

```suggestion
\[This is comparable to a compilation pipeline, with a concise intermediate representation that nicer user-facing features "desugar" into.\]
```

Or maybe it could be reworded as something like: 
```suggestion
\[This is comparable to a compilation pipeline, where nicer user-facing features "desugar" into a concise intermediate representation.\]
```